src/main/java/conan/ui/configuration/ConanConfig.java

```diff
@@ -72,17 +72,14 @@ public String getDisplayName() {
     @Override
     public JComponent createComponent() {
         downloadConfiguration.addActionListener(actionEvent -> {
-            String url = configInstallUrl.getText();
-            if (StringUtils.isBlank(url)) {
-                url = PersistencyUtils.getValue(CONFIG_INSTALL_URL);
+            String location = configInstallLocation.getText();
+            if (StringUtils.isBlank(location)) {
+                location = PersistencyUtils.getValue(CONFIG_INSTALL_LOCATION);
             }
-            PersistencyUtils.setValue(CONFIG_INSTALL_URL, url);
-            if (!Utils.validateUrl(url)) {
-                setConfigInstallRes("Bad URL", false);
-                return;
-            }
-            runConfigInstall(url);
+            PersistencyUtils.setValue(CONFIG_INSTALL_LOCATION, location);
```

Comment:
We can validate it is a valid URL OR that the path exists. Easy and the user will get that early feedback:

![image](https://user-images.githubusercontent.com/1406456/63521969-c609ad80-c4f7-11e9-8ef9-6ae22fa6a5ce.png)

Reply:
I used a catch-all message (Bad URL or path) because it could be hard to tell if the user meant to  input a path or a URL if both validations fail (valid URL or path exists).